mitmproxy/proxy/protocol/http.py

```diff
@@ -289,9 +289,12 @@ def _process_flow(self, f):
             f.request = None
             f.error = flow.Error(str(e))
             self.channel.ask("error", f)
-            raise exceptions.ProtocolException(
-                "HTTP protocol error in client request: {}".format(e)
-            ) from e
+            if not isinstance(e, exceptions.HttpSyntaxException):
```

Comment:
It's not clear from the PR when this is still triggered. What's the story here? Can we maybe add an explaining comment?